src/connectors/tagService.ts

```diff
@@ -685,7 +684,48 @@ export class TagService extends BaseService {
     includeAuthorTags?: boolean
     viewerId?: string | null
   }) => {
-    return { nodes: [], totalCount: 0 }
+    const _key =
+      key.startsWith('#') || key.startsWith('＃') ? key.slice(1) : key
+
+    if (!_key) {
+      return { nodes: [], totalCount: 0 }
+    }
+
+    const mattyChoiceTagIds = environment.mattyChoiceTagId
+      ? [environment.mattyChoiceTagId]
+      : []
+
+    // search from original tags but return duplicate free tags
+    const queryIds = this.knex('search_index.tag')
```

Comment:
For performance reason, shall we `.limit` the query results?

Reply:
limit do be set in L723？

Reply:
But the query performance of L702 impacted by `queryIds` right?

Reply:
Yes， but if `queryIds` is limited,  the final search result will lack some hit items?